src/test/java/roomescape/controller/ReservationControllerTest.java

```diff
@@ -0,0 +1,86 @@
+import roomescape.repository.fake.FakeReservationTimeRepository;
+
+class ReservationControllerTest {
+    FakeReservationRepository fakeReservationRepository = new FakeReservationRepository();
+    FakeReservationTimeRepository fakeReservationTimeRepository = new FakeReservationTimeRepository();
+    private final ReservationController reservationController = new ReservationController(fakeReservationRepository, fakeReservationTimeRepository);
+
+    private void makeStubReservationTime(Long id, LocalTime startAt) {
+        fakeReservationTimeRepository.createReservationTime(new ReservationTime(id, startAt));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {"히스타, 2002-12-02, 1, 18:00"})
+    void create(String nickname, LocalDate date, Long timeId, LocalTime time) {
+        // given
+        makeStubReservationTime(timeId, time);
+        Reservation reservation = new Reservation(null, nickname, date, timeId);
+
+        // when
+        ResponseEntity<ReservationResponse> actualResponse = reservationController.create(reservation);
+
+        // then
+        Assertions.assertThat(actualResponse).isNotNull();
+        Assertions.assertThat(actualResponse.getBody()).isNotNull();
+        assertAll(
+                () -> Assertions.assertThat(actualResponse.getStatusCode().is2xxSuccessful()).isTrue(),
+                () -> Assertions.assertThat(actualResponse.getBody().name()).isEqualTo(reservation.getName()),
+                () -> Assertions.assertThat(actualResponse.getBody().date()).isEqualTo(reservation.getDate()),
+                () -> Assertions.assertThat(actualResponse.getBody().reservationTimeResponse().id()).isEqualTo(timeId),
+                () -> Assertions.assertThat(actualResponse.getBody().reservationTimeResponse().startAt()).isEqualTo(time)
+        );
+    }
+
+    @Test
+    void read() {
+        // given
+        LocalDate givenDate = LocalDate.now();
+        LocalTime givenTime = LocalTime.now();
+        String givenName = "히스타";
+        Long givenTimeId = 1L;
+
+        create(givenName, givenDate, givenTimeId, givenTime);
+
+        // when
+        ResponseEntity<List<ReservationResponse>> actualResponse = reservationController.read();
+
+        // then
+        List<ReservationResponse> reservationList = actualResponse.getBody();
+
+        Assertions.assertThat(reservationList).hasSize(1);
+        assertAll(
+                () -> Assertions.assertThat(reservationList.getFirst().name()).isEqualTo(givenName),
+                () -> Assertions.assertThat(reservationList.getFirst().date()).isEqualTo(givenDate),
+                () -> Assertions.assertThat(reservationList.getFirst().reservationTimeResponse().id()).isEqualTo(givenTimeId),
+                () -> Assertions.assertThat(reservationList.getFirst().reservationTimeResponse().startAt()).isEqualTo(givenTime)
+        );
+    }
+
+    @Test
+    void delete() {
+        // given
+        create("히스타", LocalDate.now(), 1L, LocalTime.now());
```

Comment:
```suggestion
        create("히스타", LocalDate.now(), 2L, LocalTime.now());
```
왜 timeId를 바꾸면 테스트가 실패할까요~?

Reply:
`create()` 메서드가 항상 `1L`의 `id`로 `ReservationTime`을 생성하기 때문입니다.
테스트 간의 결합이 안좋은 이유를 바로 설명해주셨네요. 설득되었습니다! 감사합니다 :)

Reply:
테스트라는 환경상 임의의 ID로 저장하는 경우가 분명 존재할 텐데요. ID를 직접 지정할 수 없도록 하는 것이 과연 옳은지 궁금해요~

Reply:
안 그래도 해당 부분에 대해, **"테스트 코드니까 ID(pk)를 직접 지정할 수 있도록 하면 더 유연하지 않을까?"** 라는 고민을 했었는데요.

1. 테스트 환경이지만 `FakeRepository` 객체는 실제 프로덕션 `Repository`(DB를 이용하는)와 가능한 한 유사해야 한다고 생각했습니다.
2. 그런데 프로덕션에서 임의의 ID(pk)로 저장하는 경우는 일어나지 않을 일이라고 판단했습니다. (Auto increment가 적용된 pk니까)
3. 그러면 테스트 코드에서 임의의 ID로 저장해서 테스트 해야하는 경우가 존재할까? **(= 아니요)**

따라서 ID를 직접 지정할 수 없도록 했습니다. ID를 직접 지정해야 하는 테스트를 작성해야 할 일이 있을까요?

Reply:
테스트 환경은 운영 환경의 특정 시점을 재현할 수 있어야 해요. ID를 직접 지정해야 하는 경우가, 지금 상상해보기론 다음과 같은 상황이 떠오르네요. 특정 N배수의 순서로 가입한 사용자에게 무언가 혜택을 주고자 해요. 그렇다면 가입시 부여되는 ID를 활용하면 될 것이고, ID 값이 N배수인지 확인하여 로직을 수행하면 될 거예요. 이를 테스트하고자 한다면, N배수의 ID인지 아닌지 두가지 경우의 수만 테스트하면 될텐데요. 그에 반해 히스타의 방식은 N배수의 ID를 지닌 엔티티를 만들기 위해 N배수 번째까지의 엔티티를 만들어야 해요. 이처럼 ID가 로직에 영향을 미치는 경우는, ID를 직접 지정하여 테스트를 작성하곤 해요.

지금의 요구조건으로는 히스타의 방식에 문제는 없으나, ID를 직접 지정할 수 없어야 한다는 관념을 없앴으면 하는 바램에서 질문 드려보았어요~

Reply:
그렇네요. 재현 관점에서 테스트 코드에서 ID를 지정할 수 있도록 하면 좋겠네요. 구체적인 상황 제시해주셔서 이해가 됐어요!

감사합니다 !!